File: 2nd Year/2nd Semester/MPP/PROIECT-JAVA/Laborator/src/main/java/com/example/laborator/Domain/TipProba.java
package com.example.laborator.Domain;

public enum TipProba {
    NATATIE("Natatie"),
    CICLISM("Ciclism"),
    ALERGARE("Alergare");

    private final String denumire;

    TipProba(String denumire) {
        this.denumire = denumire;
    }

    public String getDenumire() {
        return denumire;
    }

    @Override
    public String toString() {
        return denumire;
    }

    public TipProba getTipProba() {
        return this;
    }
}

File: 2nd Year/2nd Semester/MPP/PROIECT-C#/Lab1/Lab1/Domain/TipProba.cs
using System;

namespace Lab1.Domain {
    public enum TipProba {
        NATATIE,
        CICLISM,
        ALERGARE
    }

    public static class TipProbaExtensions {
        public static string GetDenumire(this TipProba tipProba) {
            switch (tipProba) {
                case TipProba.NATATIE:
                    return "Natatie";
                case TipProba.CICLISM:
                    return "Ciclism";
                case TipProba.ALERGARE:
                    return "Alergare";
                default:
                    throw new ArgumentOutOfRangeException(nameof(tipProba), tipProba, null);
            }
        }
    }
}